src/components/lib/cos_kernel_api.c

```diff
@@ -149,6 +158,62 @@ __untyped_bump_alloc(struct cos_compinfo *ci)
 	return __mem_bump_alloc(ci, 1, 0);
 }
 
+static int
+__cos_mem_alias_at(struct cos_compinfo *dstci, vaddr_t dst, struct cos_compinfo *srcci, vaddr_t src, int superpage)
+{
+	int order;
+	assert(srcci && dstci);
+
+	order = (superpage) ? SUPER_PAGE_ORDER : PAGE_ORDER;
+
+	if (call_cap_op(srcci->pgtbl_cap, CAPTBL_OP_CPY, src, dstci->pgtbl_cap, dst, order)) BUG();
+
+	return 0;
+}
+
```

Comment:
If you want this PR to be accepted, I'd like to know the plan for making this a strong basis for the architecture-specific stuff.

We need a generic way to describe the superpages (as described above), but all of this code is hacked together around 2 page sizes.  What is the plan here?  Is the next PR going to include that generic layer?  Why do this PR without it?